pkg/kubelet/util/util_windows.go

```diff
@@ -71,12 +72,12 @@ func GetAddressAndDialer(endpoint string) (string, func(addr string, timeout tim
 	return "", nil, fmt.Errorf("only support tcp and npipe endpoint")
 }
 
-func tcpDial(addr string, timeout time.Duration) (net.Conn, error) {
-	return net.DialTimeout(tcpProtocol, addr, timeout)
+func tcpDial(ctx context.Context, addr string) (net.Conn, error) {
+	return (&net.Dialer{}).DialContext(ctx, tcpProtocol, addr)
 }
 
-func npipeDial(addr string, timeout time.Duration) (net.Conn, error) {
-	return winio.DialPipe(addr, &timeout)
+func npipeDial(ctx context.Context, addr string) (net.Conn, error) {
+	return winio.DialPipeContext(ctx, addr)
```

Comment:
According to https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/83003/pull-kubernetes-typecheck/1176050851903442945, `winio.DialPipeContext` doesn't exist. However, since I'm seeing it in master on `winio`, I'm guessing we need to update our `winio` version?

Reply:
yep, vendor updated.